### ISSUE TITLE
Comment highlighting should be last in *.nanorc

### ISSUE THREAD

**Issue by Xaero252** [scopatz/nanorc#47]
Purely cosmetic issue, but all comment highlighting should be the last in each nanorc. Since nano parses these configurations line by line from top to bottom, any definitions after the comment highlighting will be highlighted even if its within a comment, which is undesirable.

**Comment by ThomasR** [comment]
Not necessarily. Think about URLs, email addresses, `FIXME`/`TODO`, meta comments like vim editor settings etc.
That is stuff you’d potentially want to highlight within comments, so you should consider it individually, depending on the file type.

**Comment by agaskins** [comment]
I'm kind of with @Xaero252 on this; highlighting in the comments is distracting and, from my experience, most people think something is wrong with your editor when they see variables and such highlighted inside a comment block. That said, I do see the merit in this for certain situations like hyperlinking, as @ThomasR mentioned.My proposed solution would be to indeed include the comment parsing at the end, as @Xaero252 initially suggested, and then add specific rules for highlighting things like links, TODO/FIXME notes, etc. even lower in the file, below the generic comment highlighting rules. This way it's the best of both worlds, and we can make sure comments, when they do get jazzed up, actually look intentional. I'd be willing to work on this if anyone else thinks it has merit, and if there's no obvious side-effects to this method (that someone more experienced than I might be able to foresee).

**Comment by Xaero252** [comment]
I too see the merit, and can actually identify other editors highlighting based on the situation as well. Hyperlinks and FIXME / TODO was an excellent usage cause for highlighting within comments, Currently all sorts of strange things get highlighted within comments, which looks tacky and unintentional. Again, it's purely cosmetic, and some people may actually desire such functionality.
